### PR TITLE
Fix table scan results with disjoint pushdown filters

### PR DESCRIPTION
**Pull request by bhhari**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java [hunk 1]
```diff
@@ -1756,25 +1756,33 @@ public ConnectorPushdownFilterResult pushdownFilter(ConnectorSession session, Co
...
-                        .transform(HiveMetadata::toSubfield)
-                        .getDomains()
-                        .orElse(ImmutableMap.of()))
-                .putAll(decomposedFilter.getTupleDomain()
-                        .transform(subfield -> !isEntireColumn(subfield) ? subfield : null)
-                        .getDomains()
-                        .orElse(ImmutableMap.of()))
-                .build());
+        if (decomposedFilter.getTupleDomain().isNone()) {
+            domainPredicate = TupleDomain.none();
```
Comment (highker) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: How about
```java
return new ConnectorPushdownFilterResult(getTableLayout(session, currentLayoutHandle.get()), FALSE_CONSTANT);
```
Haven't checked if the above path can work; but looks reasonable

Comment (highker) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: NVM, let's fix this in the planner

**Comment by bhhari** [comment]
> I think this needs to happen at plan optimizer level so that we produce empty `ValuesNode` instead of a table scan going through a false filter. Could you fix this in `pushPredicateIntoTableScan`? You will need a `RowExpressionInterpreter` to interpret the value of `translatedPredicate`; if the result is false, return an empty `ValuesNode`

updated the PickTableLayout to use the interpreter, but its sufficient. It can address some constant expressions but we still need the logic in the HiveMetadata to short circuit filters like orderkey=1 and orderkey=2

**Comment by highker** [comment]
@bhhari, nice observation! Let me think about it.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java [hunk 2]
```diff
@@ -295,11 +295,14 @@ public static PlanNode pushPredicateIntoTableScan(
             SqlParser parser,
             ExpressionDomainTranslator domainTranslator)
     {
+
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java]: remove this line.

**Comment by bhhari** [comment]
fixes #13533 #13534

**Pull request merged**